src/app/features/settings/go-pro/go-pro-media-list-on-camera/go-pro-media-list-on-camera.component.ts

```diff
@@ -64,21 +69,34 @@ export class GoProMediaListOnCameraComponent implements OnInit {
       this.fetchingFilesError = undefined;
       this.fetchingFiles = true;
       this.allMediaFiles = await this.goProMediaService.getFilesFromGoPro();
-      this.fetchingFiles = false;
     } catch (error: any) {
-      this.fetchingFilesError = error.toString();
+      this.fetchingFilesError = 'Failed to fetch media from GoPro';
+      if (this.platform.is('ios')) {
+        this.fetchingFilesError =
+          'Please check iOS Settings > Capture > Local Network, make sure the permission of Local Network is allowed Capture app.';
+      }
       this.allMediaFiles = [];
+    } finally {
       this.fetchingFiles = false;
     }
   }
 
   async connectToGoProWifi() {
     try {
+      this.isConnectingToGoProWifi = true;
+
+      if (!(await this.goProBluetoothService.getConnectedDevice())) {
+        this.errorService.toastError$('Connect to GoPro via bluetooth first');
```

Comment:
```suggestion
        await this.errorService.toastError$('Connect to GoPro via bluetooth first').toPromise();
```

`toastError$` returns an Observable so I think you need conversions like this to make it work